contentcuration/contentcuration/static/js/edit_channel/exercise_creation/views.js

```diff
@@ -356,18 +356,21 @@ var EditorView = BaseViews.BaseView.extend({
     render_editor: function() {
         var self = this;
         this.toggle_loading(true);
-        this.parse_content(this.model.get(this.edit_key)).then(function(result){
-            var html = self.view_template({content: result}, {
-                data: self.get_intl_data()
+        var value = this.model.get(this.edit_key);
+        if (value && value.trim() !== "") {
```

Comment:
The code you linked to makes the assumption that value is a string. It isn't in this case (it's false), that's why we get the error. I know we can do just `value.trim()` but I think it's better to be explicit that we're checking that the result isn't a blank string.

Also, I don't want to go the regex route. :D

Reply:
the intent might be clearer then if we wrote something like this:

```
if (typeof value === "string" && value)
```

That seems more readable for me. I don't trust my JS-foo though, so I'm not sure if that's idiomatic. I'll leave this change up to you.

Reply:
I think the disconnect here is a difference in idioms between JS and Python. In JS, things cast automatically, and just about everything casts to string. That annoying need to sometimes do === instead of == is because of this; the extra = tells it not to cast before comparing. This is an area where Python focused on safe coding, while JS focused on keeping the page loading. ;-)